tests/orion/api/test_task_runs.py

```diff
@@ -77,7 +77,11 @@ async def test_create_task_run_with_state_sets_timestamp_on_server(
             json=schemas.actions.TaskRunCreate(
                 flow_run_id=flow_run.id,
                 task_key="a",
-                state=states.Completed(timestamp=pendulum.now().add(months=1)),
+                state=schemas.actions.StateCreate(
+                    type=schemas.states.StateType.COMPLETED,
+                    # This is no longer allowed
+                    # timestamp=pendulum.now().add(months=1),
```

Comment:
I just realized this test doesn't exactly test what it sounds like.  I think the assertion at the end should be:

```
start_of_test = pendulum.now()

... do the POST ...

assert start_of_test <= task_run.state.timestamp <= pendulum.now()
```

Right?

Reply:
Ah that was a holdover from when I was going to ban timestamps but I realized I had to keep them for backwards compatibility. I've updated this test to include the timestamp again and changed the name for clarity.

We're sending a timestamp with +1 month so we can do a simpler check. We could include a lower bound but I'm not sure what it adds?

Reply:
Let me know if you have additional concerns; this is how the tests were structured before so I'll leave it as is here.